src/pyfme/utils/trimmer.py

```diff
@@ -0,0 +1,241 @@
+# -*- coding: utf-8 -*-
+"""
+Python Flight Mechanics Engine (PyFME).
+Copyright (c) AeroPython Development Team.
+Distributed under the terms of the MIT License.
+
+Trimmer
+-------
+This module solves the problem of calculating the values of the state and
+control vectors that satisfy the state equations of the aircraft at the initial
+condition (or any other given condition. This cannot be done analytically
+because of the very complex functional dependence of the erodynamic data.
+Instead, it must be done with a numerical algorithm which iteratively adjusts
+the independent variables until some solution criterion is met.
+"""
+from warnings import warn
+import numpy as np
+from math import sqrt, sin, cos, tan, atan
+from scipy.optimize import least_squares
+
+from pyfme.utils.coordinates import wind2body
+from pyfme.environment.isa import atm
+
+
+def steady_state_flight_trim(aircraft, h, TAS, gamma=0, turn_rate=0,
+                             dyn_eqs=None, verbose=0):
+    """Finds a combination of values of the state and control variables that
+    correspond to a steady-state flight condition. Steady-state aircraft flight
+    can be defined as a condition in which all of the motion variables are
+    constant or zero. That is, the linear and angular velocity components are
+    constant (or zero), and all acceleration components are zero.
+
+    Parameters
+    ----------
+        aircraft : aircraft class
+            Aircraft class with methods get_forces, get_moments
+        h : float
+            Geopotential altitude for ISA (m).
+        TAS : float
+            True Air Speed (m/s).
+        gamma : float, optional
+            Flight path angle (rad).
+        turn_rate : float, optional
+            Turn rate, d(psi)/dt (rad/s).
+
+    Returns
+    -------
+        lin_vel : float array
+            [u, v, w] air linear velocity body-axes (m/s).
+        ang_vel : float array
+            [p, q, r] air angular velocity body-axes (m/s).
+        theta : float
+            Pitch angle (rad).
+        phi : float
+            Bank angle (rad).
+        alpha : float
+            Angle of attack (rad).
+        beta : float
+            Sideslip angle (rad).
+        control_vector : array_like
+            [delta_e, delta_ail, delta_r, delta_t].
+
+    Notes
+    -----
+    See section 3.4 in [1] for the algorithm description.
+    See section 2.5 in [1] for the definition of steady-state flight condition.
+
+    References
+    ----------
+    .. [1] Stevens, BL and Lewis, FL, "Aircraft Control and Simulation",
+        Wiley-lnterscience.
+    """
+
+    if dyn_eqs is None:
+        from pyfme.models.euler_flat_earth import linear_and_angular_momentum_eqs
+        dynamic_eqs = linear_and_angular_momentum_eqs
+
+    # TODO: try to look for a good inizialization method
```

Comment:
@JuanMatSa This is one of the things that is still pending: a good initialization. Maybe there is some kind of simplified analytical approach to the problem